File: solutions/javascript/scrabble-score/1/scrabble-score.js
const LETTER_SCORES = {};

const addLetterScores = (letterString, score) => {
  letterString.split('').forEach((letter) => (LETTER_SCORES[letter] = score));
};

addLetterScores('AEIOULNRST', 1);
addLetterScores('DG', 2);
addLetterScores('BCMP', 3);
addLetterScores('FHVWY', 4);
addLetterScores('K', 5);
addLetterScores('JX', 8);
addLetterScores('QZ', 10);

export const score = (word) => {
  return [...word.toUpperCase()].reduce(
    (total, letter) => total + LETTER_SCORES[letter],
    0
  );
};

File: solutions/python/scrabble-score/1/scrabble_score.py
def score(word):
    total = 0
    for char in word.upper():
        if char in "AEIOULNRST": 
            total += 1
        if char in "DG": 
            total += 2
        if char in "BCMP":
            total += 3
        if char in "FHVWY":
            total += 4
        if char == "K":
            total += 5
        if char in "JX":
            total += 8
        if char in "QZ":
            total += 10
    
    return total
